### ISSUE TITLE
Match cannot be analyzed and produces fatal error (memory issue)

### ISSUE THREAD

**Issue by RobertEcker** [nunomaduro/phpinsights#470]
| Q                | A
| ---------------- | -----
| Bug report?      | yes
| Feature request? | no
| Library version  | dev-master

```
        $test = 1;
        $array = [
            0 => ['a' => 1, 'b' => 8],
            1 => ['a' => 9, 'b' => 21],
        ];
        $index = match ($test) {
            'a' => 0,
            'b' => 1,
            default => 1,
        };
        return random_int($array[$index]['a'], $array[$index]['b']);
```

It cannot be analyzed and produces out of memory "Fatal error: Allowed memory size of ..."

**Comment by osteel** [comment]
Not sure if related but PHP Insights recently started to flag the semicolons after `match` statements as useless. Using `dev-master` as well.

**Comment by Jibbarth** [comment]
Thanks for reporting @RobertEcker, but I was unable to reproduce an out of memory error. Do you have more details ? 

@osteel, you are completly right, good catch. 

It seems that is already fixed in Slevomat/coding-standard (https://github.com/slevomat/coding-standard/issues/1171). 

But it's released under ^7.0, so we have to support this version now :slightly_smiling_face:

**Comment by RobertEcker** [comment]
```
<?php
final class Autoloader
{
    public function register(): void
    {
        spl_autoload_register(
            function (string $className): void
            {
                $names = $this->getNames($className);
            }
        );
    }

    private function getNames(string $className): array
    {
        $array = [
            0 => ['fileEndingPart' => 'api', 'len' => 3],
        ];
        $index = match (true) {
            str_ends_with($className, 'Api') => 0,
            default => 2,
        };
        return [];
    }
}
```

I tried out a minimal example (see above). I guess the "match" is the problem.
It produces the following error in the php_error.log (after some GBs, I stopped ;-) ):

PHP Warning:  Undefined array key "scope_closer" in C:\wamp\www\rob.ec\hp\lib\misc\vendor\nunomaduro\phpinsights\vendor\slevomat\coding-standard\SlevomatCodingStandard\Helpers\UseStatementHelper.php on line 195
